src/java.compiler/share/classes/javax/tools/StandardLocation.java

```diff
@@ -109,6 +111,22 @@ public enum StandardLocation implements Location {
      */
     PATCH_MODULE_PATH;
 
+    /**
+     * Canonical location instances.
+     */
+    private static final ConcurrentHashMap<String, Location> LOCATIONS = new ConcurrentHashMap<>();
+
+    private static class LazyPatternHolder {
```

Comment:
Should we keep this as a local class in `computeIsModuleOrientedLocation` as it is only used in that method??

Reply:
Even better, we can move this static utility `computeIsModuleOrientedLocation` into the `new Location` anonymous class and convert it to a local class; that way we define one fewer class as well.

Reply:
Yeah, it does not work, really. My attempts in creating a separate class/record for location, attempts to access it from `JavaFileManager`, munging build system to compile the interim javac correctly -- all this uglifies the patch. So I would like to go with the current version. I tidied up some comments, though.